worksheet_D/oxo.py

```diff
@@ -0,0 +1,115 @@
+class OxoBoard:
+    def __init__(self):
+        """ The initialiser. Initialise any fields you need here. """
+        self.oxoList = ([0,0,0],[0,0,0],[0,0,0])
+
+    def get_square(self, x, y):
+        """ Return 0, 1 or 2 depending on the contents of the specified square. """
+        return self.oxoList[x][y]
+
+    def set_square(self, x, y, mark):
+        """ If the specified square is currently empty (0), fill it with mark and return True.
+            If the square is not empty, leave it as-is and return False. """
+        if self.oxoList[x][y] == 0:
+            self.oxoList[x][y] = mark
+            return True
+        else:
+            return False
+
+    def is_board_full(self):
+        """ If there are still empty squares on the board, return False.
+            If there are no empty squares, return True. """
+        xPos = 0
+        yPos = 0
+        squaresFilled = True
+
+        while xPos <3:
+            while yPos < 3 :
```

Comment:
This goes into an infinite loop because you're not incrementing `xPos` or `yPos` inside the loop -- they're permanently at zero. It would be better to use `for` loops rather than `while` loops here.